### ISSUE TITLE
Scaler used incorrectly

### ISSUE THREAD

**Issue by YannickGibson** [ybabakhin/nie-pdl-template-errors#8]
# Backward pass
            if (
                cfg.environment.mixed_precision
                and len(cfg.environment.gpus)
            ):
                if itr % cfg.training.grad_accumulation == 0:
                    scaler.step(optimizer)  # type: ignore
                    scaler.update()
                scaler.scale(loss).backward()  # type: ignore
            else:
                if itr % cfg.training.grad_accumulation == 0:
                    optimizer.step()
                loss.backward()
                
The `scaler.scale(loss).backward()` should be called before `scaler.step(optimizer)`
